test/system/src/test/java/io/pravega/test/system/TransactionStressTest.java

```diff
@@ -0,0 +1,388 @@
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = writerList.stream().map(writer ->
+                    startWritingIntoTxn(eventData, writer, eventWriteCount)).collect(Collectors.toList());
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = readerList.stream().map(reader ->
+                    startReading(eventsReadFromPravega, eventWriteCount, eventReadCount, stopReadFlag, reader))
+                                                                       .collect(Collectors.toList());
+
+            //wait for writers completion
+            log.info("Wait for writers execution to complete");
+            FutureHelpers.allOf(writerFutureList).get();
+
+            //set the stop read flag to true
+            log.info("Stop read flag status {}", stopReadFlag);
+            stopReadFlag.set(true);
+
+            //wait for readers completion
+            log.info("Wait for readers execution to complete");
+            FutureHelpers.allOf(readerFutureList).get();
+
+            log.info("All writers have stopped. Setting Stop_Read_Flag. Event Written Count:{}, Event Read " +
+                    "Count: {}", eventWriteCount.get(), eventsReadFromPravega.size());
+            assertEquals(eventWriteCount.get(), eventsReadFromPravega.size());
+            assertEquals(eventWriteCount.get(), new TreeSet<>(eventsReadFromPravega).size()); //check unique events.
+
+            //close all the writers
+            log.info("Closing writers");
+            writerList.forEach(writer -> writer.close());
+            //close all readers
+            log.info("Closing readers");
+            readerList.forEach(reader -> reader.close());
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        //seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(scope, STREAM_NAME);
+        log.info("Sealing stream {}", STREAM_NAME);
+        assertTrue(sealStreamStatus.get());
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        log.info("Deleting stream {}", STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+        //delete the scope
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "MultiReaderWriterTxnWithFailOver");
+    }
+
+
+    private CompletableFuture<Void> startReading(final ConcurrentLinkedQueue<Long> readResult, final AtomicLong writeCount, final
```

Comment:
Mutable values such as readResult, writeCount, readCount, and exitFlag should not be passed as arguments to methods per our contributing guidelines.